### DISCUSSION TITLE
Validation

### DISCUSSION THREAD

**Discussion by kirmanserdar** [Project-MONAI/MONAI#2283] [Q&A]
Hi,

I am using this [this](https://github.com/Project-MONAI/tutorials/tree/master/3d_segmentation/challenge_baseline)  example for my task. But I want to see validation results for every batch. In this work SupervisedEvaluator take all 3d volume as an input and produce 1 Dice score. But my models take (192, 192, 4) as an input So I  need to calculate validation score for every (192, 192, 4) batch.

**Comment by Nic-Ma** [comment]
Hi @wyli ,

Could you please help share some comments about this question? I think you developed the mentioned example.

Thanks in advance.

**Reply by kirmanserdar** [reply]
Hi @Nic-Ma,

Yes, I changed it. But I can't fix the validation part. With `monai.egines.SupervisedEvaluator` the validation part starts but It calculate validation dice score for whole volume. I want to do it by pieces like what we do in train part. 

ıf my Volume 192 192 75 and my model produce 192 192 16 output:
In train, The model calculate loss for part of (192, 192, 16).
In validation, The model calculate dice score for whole (192, 192, 75) volume.

I want to make validation part like a train part. 

Thank you for your interest

**Reply by Nic-Ma** [reply]
Hi @kirmans ,

Thanks for your explanation, I got your feature request now.
Actually, I am working on a PR to enhance our metrics: https://github.com/Project-MONAI/MONAI/pull/2291, one enhancement of the PR is to provide different `reduction` methods. If you set `batch_mean` reduction, then you can get the dice score for every batch:
https://github.com/Project-MONAI/MONAI/pull/2291/files#diff-33cb00af31947919bb3d3dad67eed8447589ac5cb9100d722b86930e2bb13799R43

Thanks.
